shared/bridges/cctp/evm_bridge_client.go

```diff
@@ -265,3 +265,54 @@ func (c *EVMBridgeClient) OrderStatus(ctx context.Context, gatewayContractAddres
 
 	return status, nil
 }
+
+func (c *EVMBridgeClient) QueryOrderSubmittedEvent(ctx context.Context, gatewayContractAddress, orderID string) (*fast_transfer_gateway.FastTransferOrder, error) {
+	fastTransferGateway, err := fast_transfer_gateway.NewFastTransferGateway(
+		common.HexToAddress(gatewayContractAddress),
+		c.client,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	orderIDBytes, err := hex.DecodeString(orderID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create topic for OrderRefunded event to filter logs for OrderRefunded events with this orderID
+	orderSubmittedTopic := [][32]byte{[32]byte(orderIDBytes)}
+	filterOpts := &bind.FilterOpts{
+		Context: ctx,
+	}
+
+	iterator, err := fastTransferGateway.FilterOrderSubmitted(filterOpts, orderSubmittedTopic)
+	if err != nil {
+		return nil, fmt.Errorf("filtering OrderSubmitted events: %w", err)
+	}
+
+	// Find the most recent OrderRefunded event for this orderID
+	var order *fast_transfer_gateway.FastTransferOrder
+	for iterator.Next() {
+		if iterator.Event != nil {
+			decodedOrder := decodeOrder(iterator.Event.Order)
+			order = &decodedOrder
+		}
+	}
+
+	return order, nil
+}
+
+func decodeOrder(bytes []byte) fast_transfer_gateway.FastTransferOrder {
```

Comment:
the transfer monitoring already has this exact same [decode order function](https://github.com/skip-mev/skip-go-fast-solver/blob/de5ebbae08e3f64dfac8032b88a0ff2936312efa/transfermonitor/transfermonitor.go#L343). we should probably make a shared/order package and have a shared order type that wrap the type being returned here and has a decode function on it, so we don't have to maintain two